examples/documentation_examples/examples_test.go

```diff
@@ -20,12 +21,23 @@ import (
 	"go.mongodb.org/mongo-driver/internal/testutil"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/description"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
 	"go.mongodb.org/mongo-driver/mongo/options"
 	"go.mongodb.org/mongo-driver/x/bsonx"
 	"go.mongodb.org/mongo-driver/x/mongo/driver/connstring"
 	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
 )
 
+func TestMain(m *testing.M) {
```

Comment:
I think this may be necessary per package for `mtest` to work correctly. See the same initialization happening in [mongo/integration/main_test.go](https://github.com/mongodb/mongo-go-driver/blob/57747f8fe7130b6afdade2a946d399afe6ae7079/mongo/integration/main_test.go#L18) and in [mongo/integration/unified/main_test.go](https://github.com/mongodb/mongo-go-driver/blob/57747f8fe7130b6afdade2a946d399afe6ae7079/mongo/integration/unified/main_test.go#L18).

Reply:
[`TestMain`](https://pkg.go.dev/testing#hdr-Main) is typically where global setups are constructed for the testing threads.

Reply:
Gotcha; thanks for the explanation.